mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,462 @@
+                        loaded_model = mlflow.pyfunc.load_model(logged_model.model_uri)
+                        # Predict on a flow input dictionary.
+                        print(loaded_model.predict({"text": "Python Hello World!"}))
+
+        :param params: Additional parameters to pass to the model for inference.
+
+                       .. Note:: Experimental: This parameter may change or be removed in a future
+                                               release without warning.
+
+        :return: Model predictions. Dict type, example ``{"output": "\n\nprint('Hello World!')"}``
+        """
+        if isinstance(data, pd.DataFrame):
+            messages = data.to_dict(orient="records")
+            if len(messages) > 1:
+                raise mlflow.MlflowException.invalid_parameter_value(
+                    _INVALID_PREDICT_INPUT_ERROR_MESSAGE
+                )
+            messages = messages[0]
+        elif isinstance(data, dict):
+            messages = data
+        else:
+            raise mlflow.MlflowException.invalid_parameter_value(
+                _INVALID_PREDICT_INPUT_ERROR_MESSAGE
+            )
+        return self.model_invoker.invoke(messages)
+
+
+def _load_pyfunc(path, model_config: Optional[Dict[str, Any]] = None):
+    """
+    Load PyFunc implementation for Promptflow. Called by ``pyfunc.load_model``.
+    :param path: Local filesystem path to the MLflow Model with the ``promptflow`` flavor.
+    """
+    from promptflow import load_flow
+
+    model_flow_path = os.path.join(path, _MODEL_FLOW_DIRECTORY)
+    model = load_flow(model_flow_path)
+    return _PromptflowModelWrapper(model=model, model_config=model_config)
+
+
+@experimental
+def load_model(model_uri, dst_path=None):
+    """
+    Load a Promptflow model from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model. For example:
+
+                      - ``/Users/me/path/to/local/model``
+                      - ``relative/path/to/local/model``
+                      - ``s3://my_bucket/path/to/model``
+                      - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+
+                      For more information about supported URI schemes, see
+                      `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
+                      artifact-locations>`_.
+    :param dst_path: The local filesystem path to which to download the model artifact.
+                     This directory must already exist. If unspecified, a local output
+                     path will be created.
+
+    :return: A Promptflow model instance
+    """
+    from promptflow import load_flow
+
+    local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
+    model_data_path = os.path.join(local_model_path, _MODEL_FLOW_DIRECTORY)
+    model = load_flow(model_data_path)
+    setattr(model, "predict", _PromptflowModelWrapper(model).predict)
```

Comment:
Do we really need`predict` API? `load_model()` API is intended to load the underlying model, if users want to have `predict` API then they should use pyfunc load. Also I don't think we want users to write production code relying on this patch:)

Reply:
Got it, removed.